### PR TITLE
Pick river/road/rail/maglev sprite depending on terrain class (closes #2232)

### LINKED ISSUE #2232: Change river/road/rail/maglev shape depending on terrain class
Related to #786, but not quite the same.

### PR DESCRIPTION
**Pull request by hugomflavio**

### CODE REVIEW

**Review by lmoureaux — CHANGES_REQUESTED** [review]
I think we need a new tileset feature flag for this. They're defined at the top of `tilespec.cpp`:
```cpp
#define TILESPEC_CAPSTR                                                     \
  "+Freeciv-tilespec-Devel-2019-Jul-03 duplicates_ok precise-hp-bars "      \
  "unlimited-unit-select-frames unlimited-upkeep-sprites hex_corner"
```
The documentation page listing them should be updated as well, in `docs/Modding/Tileset/compatibility.rst`.

**Review by lmoureaux — APPROVED** [review]

File: docs/Modding/Tilesets/compatibility.rst [hunk 1]
```diff
@@ -43,3 +43,10 @@ unlimited-upkeep-sprites
 
 hex_corner
     Support for this option signals the availability of the new ``hex_corner`` sprite type for terrain.
+
+terrain-specific-extras
+    Terrain-specific extra sprites allow for different sprites to be used depending on the terrain. Currently,
+    only rivers/roads/rails/maglev sprites may be made terrain-specific, but compatibility for specials and
+    bases is expected to roll out soon. To use terrain-specific sprites, use the same nomenclature as before,
+    but with the terrain time included after the type of extra.
```
Comment (blabber) [inline docs/Modding/Tilesets/compatibility.rst:51]: "terrain time" sounds wrong. Should this be "terrain kind" or "terrain tile"?

**Comment by hugomflavio** [comment]
Yep, should be terrain name actually. I can fix that once we tackle the
specials and bases. Unless it needs to be fixed right away?

**Comment by lmoureaux** [comment]
A small PR fixing it won't hurt. This way we don't forget

**Pull request merged**